emcc.py

```diff
@@ -1132,13 +1132,19 @@ def check(input_file):
       shared.Settings.FILESYSTEM = 0
       shared.Settings.FETCH = 1
       options.js_libraries.append(shared.path_from_root('src', 'library_asmfs.js'))
+      # library_asmfs.js requires emscripten_asmfs_readdir
+      shared.Settings.EXPORTED_FUNCTIONS += ['_emscripten_asmfs_readdir']
+      shared.Settings.EXPORTED_FUNCTIONS += ['___errno_location']
 
     if shared.Settings.FETCH and final_suffix in JS_CONTAINING_ENDINGS:
       input_files.append((next_arg_index, shared.path_from_root('system', 'lib', 'fetch', 'emscripten_fetch.cpp')))
       next_arg_index += 1
       options.js_libraries.append(shared.path_from_root('src', 'library_fetch.js'))
       if shared.Settings.USE_PTHREADS:
         shared.Settings.FETCH_WORKER_FILE = unsuffixed(os.path.basename(target)) + '.fetch.js'
+        # This is a hack for make_fetch_worker(), which requires
+        # emscripten_is_main_runtime_thread, which is called from emscripten_fetch()
+        shared.Settings.EXPORTED_FUNCTIONS += ['_emscripten_fetch']
```

Comment:
I don't understand why this change would be needed to make the tests pass for this change.  Can you explain a bit more how this is related?

Reply:
Sure, it was non-obvious to me for a few hours as well :) In the ASMFS/USE_PTHREADS/test_readdir.c test in test_browser.py, the only FS code that is used is opendir() and friends. This causes `opt` to dead code eliminate `static open()` in asmfs.cpp, and in turn removing the only dependence on `emscripten_fetch()`, in turn removing the only dependence on `emscripten_is_main_runtime_thread()`, which later causes `make_fetch_worker()` fail.

Reply:
I see.   But if its `emscripten_is_main_runtime_thread` that you need then why not depend on that?

Reply:
Just because the condition is on `Settings.FETCH` so it seemed more appropriate.